controllers/mock.go

```diff
@@ -66,68 +66,68 @@ type DBFunctionInterfaceMock struct {
 	err                  error
 }
 
-func (i DBFunctionInterfaceMock) GetPrivileges() (models.Privileges, error) {
+func (i *DBFunctionInterfaceMock) GetPrivileges() (models.Privileges, error) {
 	return i.privileges, i.err
 }
 
-func (i DBFunctionInterfaceMock) GetUser(queryString string, keyValue interface{}, tx *sql.Tx) (*models.User, error) {
+func (i *DBFunctionInterfaceMock) GetUser(queryString string, keyValue interface{}, tx *sql.Tx) (*models.User, error) {
 	return i.user, i.err
 }
 
-func (i DBFunctionInterfaceMock) AddUser(user *models.User, tx *sql.Tx) error {
+func (i *DBFunctionInterfaceMock) AddUser(user *models.User, tx *sql.Tx) error {
 	*i.userAdded = true
 	return i.err
 }
 
-func (i DBFunctionInterfaceMock) AddAsset(assetType string, asset *models.Asset, tx *sql.Tx) error {
+func (i *DBFunctionInterfaceMock) AddAsset(assetType string, asset *models.Asset, tx *sql.Tx) error {
 	return i.err
 }
 
-func (i DBFunctionInterfaceMock) DeleteAsset(assetType string, assetID *uuid.UUID, tx *sql.Tx) error {
+func (i *DBFunctionInterfaceMock) DeleteAsset(assetType string, assetID *uuid.UUID, tx *sql.Tx) error {
 	return i.err
 }
 
-func (i DBFunctionInterfaceMock) AddProductUsers(productID *uuid.UUID, productUsers models.ProductUsers, tx *sql.Tx) error {
+func (i *DBFunctionInterfaceMock) AddProductUsers(productID *uuid.UUID, productUsers models.ProductUsers, tx *sql.Tx) error {
 	return i.err
 }
 
-func (i DBFunctionInterfaceMock) AddProduct(product *models.Product, tx *sql.Tx) error {
+func (i *DBFunctionInterfaceMock) AddProduct(product *models.Product, tx *sql.Tx) error {
 	*i.productAdded = true
```

Comment:
If the func. receiver is pointer, then here you try to change the value of the pointer of the pointer. :D
You should remove asterisks from the method body.

Reply:
Will raise a ticket to fix after funtional test task is completed